logstash-core/lib/logstash/config/source_loader.rb

```diff
@@ -93,6 +93,12 @@ def fetch
 
           SuccessfulFetch.new(pipeline_configs)
         rescue => e
+          # es-output 12.0.2 throws 404 as error
+          if e.is_a?(LogStash::Outputs::ElasticSearch::HttpClient::Pool::BadResponseCodeError) && e.response_code == 404
+            logger.error("No configuration found in the configured sources.")
+            return SuccessfulFetch.new({})
```

Comment:
Should this be `Array` instead of `Hash`? 

Mocking it in tests uses Array it looks like https://github.com/elastic/logstash/blob/5573b5ad777eef7a4204f2376bd601090d7866bc/logstash-core/spec/support/mocks_classes.rb#L231

```suggestion
            return SuccessfulFetch.new([])
```